File: JavaScript/NoPrefixSet.js
// There is a given list of strings where each string contains only lowercase letters from [a-j], inclusive. The set of strings is said to be a GOOD SET if no string is a prefix of another string. In this case, print GOOD SET. Otherwise, print BAD SET on the first line followed by the string being checked.

function noPrefix(words) {
    const length = words[0]
    words = words.splice(1,words.length)
    words.sort((a, b) => a.length - b.length);

    for (let i = 0; i < length; i++){
        let temp = words[i]
        for (let j=1; j < length; j++) {
            if (temp === words[j].substring(0, temp.length)) {
                console.log("BAD SET")
                console.log(words[j])
                return
            }
        }
    }

    console.log("GOOD SET")
}

export {noPrefix}

File: Python/NoPrefixSet.py
# // There is a given list of strings where each string contains only lowercase letters from [a-j], inclusive. The set of strings is said to be a GOOD SET if no string is a prefix of another string. In this case, print GOOD SET. Otherwise, print BAD SET on the first line followed by the string being checked.

def noPrefix(words):
    length = int(words[0])
    words = words[1:len(words)]
    words.sort();

    for i in range(length):
        temp = words[i]
        for j in range(1, length):
            if (temp == words[j][0:len(temp)]):
                print("BAD SET")
                print(words[j])
                return
        
    print("GOOD SET")
    return